src/app/shared/components/charts/chart-average-commits-daily/chart-average-commits-daily.component.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {UserQuery} from '../../../models/user-query.model';
+import {ChartDataSets, ChartOptions, ChartType} from 'chart.js';
+import {Label} from 'ng2-charts';
+
+@Component({
+  selector: 'app-chart-average-commits-daily',
+  templateUrl: './chart-average-commits-daily.component.html',
+  styleUrls: ['./chart-average-commits-daily.component.scss']
+})
+export class ChartAverageCommitsDailyComponent implements OnInit {
+
+  @Input() userQueries: UserQuery[];
+
+  barChartOptions: ChartOptions = { responsive: true, maintainAspectRatio: false };
+  barChartLabels: Label[] = [];
+  barChartType: ChartType = 'bar';
+  barChartLegend = true;
+  barChartData: ChartDataSets[] = [];
+
+  constructor() { }
+
+  ngOnInit(): void {
+    const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const contributionsPerWeekday = [0, 0, 0, 0, 0, 0, 0];
+    const weekDayCounter = [0, 0, 0, 0, 0, 0, 0];
+
+    for (const query of this.userQueries) {
+      const weeks = query.user.contributionsCollection.contributionCalendar.weeks;
+
+      // counting the total amount of contributions per weekday and counting how many of each weekday there are
+      for (const week of weeks) {
+        for (const day of week.contributionDays) {
+          const dayIndex = new Date(day.date).getDay();
```

Comment:
Use the day.weekday instead of the date project

Reply:
Instruction unclear. ContributionCalendarDay does not seem to have a "weekday" attribute.